purchases/src/test/java/com/revenuecat/purchases/google/BillingWrapperTest.kt

```diff
@@ -2711,6 +2710,75 @@ class BillingWrapperTest {
 
     // endregion
 
+    @Test
+    fun `queryPurchasesAsync only calls one response when BillingClient responds twice`() {
+        var numCallbacks = 0
+
+        val slot = slot<PurchasesResponseListener>()
+        every {
+            mockClient.queryPurchasesAsync(
+                any<QueryPurchasesParams>(),
+                capture(slot)
+            )
+        } answers {
+            slot.captured.onQueryPurchasesResponse(billingClientOKResult, emptyList())
+            slot.captured.onQueryPurchasesResponse(billingClientOKResult, emptyList())
+        }
+
+        wrapper.queryPurchases(
+            appUserID = "appUserID",
+            onSuccess = {
+                numCallbacks++
+            },
+            onError = {
+                fail("shouldn't be an error")
+            }
+        )
+
+        assertThat(numCallbacks).isEqualTo(1)
+    }
+
+    @Test
+    fun `queryPurchasesAsync only calls one response when BillingClient responds twice from different threads`() {
+        val numCallbacks = AtomicInteger(0)
```

Comment:
This AtomicInteger was tricky, had to make it atomic otherwise tests failed on CI. Made the change in the other tests that are similar to this one just in case